tiledb/tests/test_libtiledb.py

```diff
@@ -471,6 +471,29 @@ def test_datetime_attribute(self):
         assert attr.dtype != np.dtype(np.datetime64("", "Y"))
         assert attr.dtype != np.dtype(np.datetime64)
 
+    @pytest.mark.parametrize("sparse", [True, False])
+    def test_ascii_attribute(self, sparse):
+        path = self.path("test_ascii")
+        dom = tiledb.Domain(
+            tiledb.Dim(name="d", domain=(1, 4), tile=1, dtype=np.uint32)
+        )
+        attrs = [tiledb.Attr(name="A", dtype="ascii", var=True)]
+        schema = tiledb.ArraySchema(domain=dom, attrs=attrs, sparse=sparse)
+        tiledb.Array.create(path, schema)
+
+        ascii_data = ["a", "b", "c", "ABC"]
+        unicode_data = ["±", "×", "÷", "√"]
+
+        with tiledb.open(path, "w") as A:
+            with self.assertRaises(tiledb.TileDBError):
+                A[np.arange(1, 5)] = unicode_data
+            A[np.arange(1, 5)] = ascii_data
+
+        with tiledb.open(path, "r") as A:
+            # print(A.schema)
+            assert A.schema.attr("A").dtype == np.bytes_
```

Comment:
Please add check of the `schema.dump` and make sure the attribute is `TILEDB_STRING_ASCII` (can use capfd, there's a few other tests that use it for checking output).